src/tron/TronEngine.ts

```diff
@@ -1008,31 +1065,54 @@ export class TronEngine extends CurrencyEngine<TronTools, SafeTronWalletInfo> {
             throw new Error('calcTxFee Failed to estimate fee')
           }
 
-          this.energyEstimateCache[`${receiverAddress}:${contractAddress}`] =
-            json.energy_used + json.energy_penalty
+          // Apply a heuristic for first time recipients.
+          // triggerconstantcontract energy estimates do not reflect first-time
+          // recipient storage cost, even though it returns recipient balances.
+          // https://www.reddit.com/r/Tronix/comments/rlhv90/tron_transaction_fee_is_different_than_predicted/
+          const adjustedEnergy = recipientHasReceivedToken
+            ? json.energy_used
+            : json.energy_used * 2.2
+          this.energyEstimateCache[cacheKey] = adjustedEnergy
+
+          this.log(
+            `Energy estimate (${
+              recipientHasReceivedToken ? 'active' : 'inactive'
+            }): ${adjustedEnergy} (raw used: ${json.energy_used}, penalty: ${
+              json.energy_penalty
+            })`
+          )
         } catch (e) {
-          this.log.warn('trx_estimateEnergy error. Using a high default.', e)
+          this.log.warn('trx_estimateEnergy error. Using a default.', e)
+          // Use different defaults based on recipient status
+          this.energyEstimateCache[cacheKey] = recipientHasReceivedToken
+            ? DEFAULT_ENERGY
+            : DEFAULT_ENERGY_NO_BALANCE
         }
+      } else {
+        this.log(
+          `Using cached energy estimate for ${cacheKey}: ${this.energyEstimateCache[cacheKey]}`
+        )
       }
 
       // Get dynamic factor for the contract if available
-      energyFactor = await this.getDynamicEnergyFactor(contractAddress)
+      const energyFactorInfo = await this.getDynamicEnergyFactor(
+        contractAddress
+      )
+      energyFactor = energyFactorInfo.factor
```

Comment:
These values are only used for logging.

Reply:
Yeah, I want to make sure we maintain logs in case something goes wrong in the future, we don't have to try to somehow reproduce. I.e. when fees spike again or some other token that doesn't work with the 2.2 heuristic